File: Python/07 - Pilas y colas/Miguel753Dav9.py
# Pila/Stack (LIFO)

stack = []
stack.append(1) # push
stack.append(2) # push
stack.append(3) # push
print(stack)
# pop
stack_item = stack[len(stack) - 1]
del stack[len(stack) - 1]
print(stack.pop())
print(stack_item)
print(stack)

# Cola/Queue (FIFO)
queue = []

# enqueue
queue.append(1)
queue.append(2)
queue.append(3)

print(queue)

# dequeue
queue_item = queue[0]

del queue[0]

print(queue_item)

print(queue.pop(0))

print(queue)

File: Java/src/Topic07_StacksAndQueues/Miguel753Dav9.java
import java.util.Stack;
import java.util.Queue;
import java.util.ArrayDeque;

public class Miguel753Dav9 {
    public static void main(String[] args){
        Stack<Integer> pila = new Stack<>();
        pila.push(1);
        pila.push(2);
        pila.push(3);
        pila.push(4);
        pila.push(5);
        System.out.println(pila);
        System.out.println(pila.pop());
        System.out.println(pila);

        Queue<Integer> cola = new ArrayDeque<>();

        cola.add(1);
        cola.add(2);
        cola.add(3);
        cola.add(4);
        cola.add(5);
        System.out.println(cola);
        System.out.println(cola.poll());
        System.out.println(cola);
    }
}
